test-utils/src/templates/query_test_header_template

```diff
@@ -11,30 +11,28 @@ use mongosql::{{
     options::{{ExcludeNamespacesOption, SqlOptions}},
     Translation,
 }};
-use std::{{collections::HashSet, fs, io::Read, path::PathBuf, string::ToString}};
-
-use std::sync::Once;
-
+use sql_engines_common_test_infra::{{parse_yaml_test_file, YamlTestFile}};
+use std::{{collections::HashSet, fs, io::Read, path::PathBuf, string::ToString, sync::Once}};
 use test_utils::*;
 
 static INIT: Once = Once::new();
 
-pub fn initialize_test() -> (&'static Client, &'static QueryYamlTestFile, &'static Catalog) {{
+pub fn initialize_test() -> (&'static Client, &'static YamlTestFile<QueryTestCase>, &'static Catalog) {{
     static mut CLIENT: Option<Client> = None;
-    static mut TEST_FILE: Option<QueryYamlTestFile> = None;
+    static mut TEST_FILE: Option<YamlTestFile<QueryTestCase>> = None;
     static mut CATALOG: Option<Catalog> = None;
 
     INIT.call_once(|| {{
         unsafe {{
             CLIENT = Some(Client::with_uri_str(MONGODB_URI.clone()).map_err(Error::CannotCreateMongoDBClient).unwrap());
-            let test_file = parse_query_yaml_file(PathBuf::from({path})).unwrap();
-            let db_names = test_file.tests.iter().flat_map(|t| {{
+            let test_file = parse_yaml_test_file("{path}").unwrap();
```

Comment:
nit -- do we still need double curlys here?

Reply:
Can you clarify which braces you're asking about? The line the comment is associated with implies you are asking about `{path}`, is that right? If so, this had single-braces before this pr and continues to have them now as part of this one. The single-braces mean the entire expression should be replaced with its named mapping, in this case `"path"`.

If you mean other braces, it seems to me all of the double-ones are appropriate. I don't see any instances of `{{` or `}}` in any of the generated files.